### PR TITLE
Feature/#13 build schema

### PR DESCRIPTION
**Pull request by cedrickmandocdoc**
Added basic schema and graphql crud. After this one I will be creating migration ticket. I think we will not rely fully on the migration generate of TypeORM

### CODE REVIEW

**Comment by cedrickmandocdoc** [comment]
I think this PR covers these tickets: #13 #14 #15

File: src/app/appointment/appointment.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+export class AppointmentResolver {
+  constructor(
+    @Inject('AppointmentRepositoryToken')
+    private readonly appointmentRepository: AppointmentRepository,
+    private readonly appointmentService: AppointmentService
+  ) {}
+
+  @Query()
+  async getAppointment(request, args) {
+    return this.appointmentRepository.findOne(args.id);
```
Comment (isotopeee) [inline src/app/appointment/appointment.resolver.ts]: I think it's cleaner to use the `destructuring assignment`  when accessing the properties from `args`.

File: src/app/appointment/appointment.entity.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  OneToOne,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn
+} from 'typeorm';
+import { Doctor } from '../doctor';
+import { Patient } from '../patient';
+
+@Entity()
+export class Appointment {
+  @PrimaryGeneratedColumn() public id?: number;
```
Comment (jmaicaaan) [inline src/app/appointment/appointment.entity.ts]: Please break into a new line
```
  @PrimaryGeneratedColumn() 
  public id?: number;
```
And also, should that be optional?

File: src/app/role/role.entity.ts [hunk 3]
```diff
@@ -1,8 +1,26 @@
...
+  Column,
+  CreateDateColumn,
+  Entity,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn
+} from 'typeorm';
 
 @Entity()
 export class Role {
   @PrimaryGeneratedColumn() public id?: number;
```
Comment (jmaicaaan) [inline src/app/role/role.entity.ts]: Same here, please put it down. And also, do we want it as optional?

File: src/app/patient/patient.entity.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  Column,
+  CreateDateColumn,
+  Entity,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn
+} from 'typeorm';
+
+@Entity()
+export class Patient {
+  @PrimaryGeneratedColumn() public id: number;
```
Comment (jmaicaaan) [inline src/app/patient/patient.entity.ts]: Same here, please put it down

File: src/app/hospital/hospital.entity.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  Column,
+  CreateDateColumn,
+  Entity,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn
+} from 'typeorm';
+
+@Entity()
+export class Hospital {
+  @PrimaryGeneratedColumn() public id?: number;
```
Comment (jmaicaaan) [inline src/app/hospital/hospital.entity.ts]: Same here, please put it down. And also, do we want it as optional?

File: src/app/doctor/doctor.entity.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  JoinTable,
+  ManyToMany,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn
+} from 'typeorm';
+import { Hospital } from '../hospital';
+
+@Entity()
+export class Doctor {
+  @PrimaryGeneratedColumn() public id: number;
```
Comment (jmaicaaan) [inline src/app/doctor/doctor.entity.ts]: Same here, please put it down.

**Comment by jmaicaaan** [comment]
[Update] base controller will not be implemented here right now. Linting configuration will go in another ticket. Merging and closing related ticket. Thank you for the PR

**Pull request merged**